### PR TITLE
add Form module to Experimental project

### PR DESCRIPTION
**Pull request by theimowski**
as discussed on Gitter channel, this is an experimental helper module to work with forms.
Feedback more than welcome ;)

### CODE REVIEW

File: src/Experimental/Form.fs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+open System
+open System.Text.RegularExpressions
+
+open Suave.Html
+open Suave.Utils
+open Suave.Types
+open Suave.Model
+
+[<Literal>]
+let emailPattern = @"\A(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)\Z"
```
Comment (haf) [inline src/Experimental/Form.fs]: There are no valid e-mail regex patterns.

You're better off using MailAddress from System.Net.Mail.

**Comment by haf** [comment]
Looks interesting

**Comment by haf** [comment]
I'm going to leave this open until @ademar has had a look.

**Comment by theimowski** [comment]
Sure, I'll correct the email valid stuff soon

**Pull request merged**